x-pack/filebeat/input/s3/input.go

```diff
@@ -567,7 +597,8 @@ func (p *s3Input) deleteMessage(queueURL string, messagesReceiptHandle string, s
 	}
 
 	req := svcSQS.DeleteMessageRequest(deleteMessageInput)
-	_, err := req.Send(p.context)
+	ctx := context.Background()
```

Comment:
I think you have p.context available. In general it's preferable to have a main context for the whole input, and inherit these from it. This way you can cancel the main context when closing the input, and that will make sure you don't leave any request pending.

Reply:
Agree, I should definitely use p.context instead here. It's probably a good idea to add timeout in p.context so all requests using it will have a timeout. This case, not only to GetObjectRequest, all API calls will timeout after a configurable amount of time. What do you think?